src/components/GrampsjsFormSelectType.js

```diff
@@ -95,40 +104,49 @@ class GrampsjsFormSelectType extends GrampsjsTranslateMixin(LitElement) {
                 `
               )}
         </mwc-select>
-      </p>
-
-      ${this.nocustom
-        ? ''
-        : html`
-            <mwc-button
-              @click="${this.switchTypeInput}"
-              ?disabled="${this.disabled}"
-            >
-              ${this._hasCustomType
-                ? this._('Switch to default type')
-                : this._('Switch to custom type')}
-            </mwc-button>
-          `}
-      ${this.nocustom || !this._hasCustomType
-        ? ''
-        : html`
-            <h4 class="label">${this._('Custom type')}</h4>
-            <p>
+        ${this.nocustom || !this._hasCustomType
+          ? ''
+          : html`
               <mwc-textfield
                 ?disabled="${this.disabled}"
                 style="width:100%"
-                ?required=${this._hasCustomType}
+                ?required=${this.required}
                 validationMessage="${this._('This field is mandatory')}"
-                @change="${this.handleChange}"
+                @input="${this.handleChange}"
+                label="${this.loadingTypes
+                  ? this._('Loading items...')
+                  : `${this.label} ${this._('Custom')}`}"
                 id="custom-type"
               >
               </mwc-textfield>
-            </p>
-          `}
+            `}
+        ${this.nocustom
+          ? ''
+          : html`
+              <mwc-icon-button
+                style="margin-left: 8px"
+                icon="${this._hasCustomType ? 'remove' : 'add'}"
+                id="button-switch-type"
+                @click="${this.switchTypeInput}"
+                ?disabled="${this.disabled}"
+              ></mwc-icon-button>
+              <grampsjs-tooltip
+                for="button-switch-type"
+                content="${this._hasCustomType
+                  ? this._('Switch to default type')
+                  : this._('Add custom type')}"
```

Comment:
I suggest to leave the string as it was orginally to not lose the translations.

Reply:
I can do that. Still I would like to propose to add the string 'Add custom type' for this scenario. It's more self explaning. But I woulf leave it like it was for now. 😄

Reply:
Or maybe I add the string. From UX-perspective I see it like this:
If I set a custom type and save it will become a default type (delivered by the backend). When users read "switch" they might expect to see the custom type set before there.

Reply:
Not sure what you mean. There are two kinds of types in Gramps: default types and custom types. The default types are translated, while the custom types are just arbitrary strings. You can either select from a list of default types, or use a custom type instead. If you want to replace a custom type by a default type again, you have to switch back to the dropdown list.

Reply:
After trying it out I get your point and I think you're right :smile: I missed the fact that the dropdown contains both custom and default types.

So: great idea.